internal/encoding/merge/parquet.go

```diff
@@ -0,0 +1,295 @@
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.INTEGER = &parquet.IntType{BitWidth: 64, IsSigned: false}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_UINT_64)
+		fieldHandler = uintHandler(64)
+	case "float64":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_FLOAT)
+		fieldHandler = floatHandler
+	case "double":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_DOUBLE)
+		fieldHandler = doubleHandler
+	case "int":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_INT64)
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.INTEGER = &parquet.IntType{BitWidth: 64, IsSigned: true}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_INT_64)
+		fieldHandler = intHandler(64)
+	case "json":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_BYTE_ARRAY)
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.JSON = &parquet.JsonType{}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_JSON)
+		fieldHandler = jsonHandler
+	default:
+		return nil, nil, fmt.Errorf("unsupported type %q", typ)
+	}
+
+	fieldHandler = optionalHandler(fieldHandler)
+
+	return col, fieldHandler, nil
+}
+
+func byteArrayHandler(s interface{}) (interface{}, error) {
+	localString := fmt.Sprintf("%v", s)
+
+	return []byte(localString), nil
+}
+
+func booleanHandler(s interface{}) (interface{}, error) {
+	localString := fmt.Sprintf("%v", s)
+
+	return strconv.ParseBool(localString)
+}
+
+func uintHandler(bitSize int) func(interface{}) (interface{}, error) {
+	return func(s interface{}) (interface{}, error) {
+		localString := fmt.Sprintf("%v", s)
+		i, err := strconv.ParseUint(localString, 10, bitSize)
+		if err != nil {
+			return nil, err
+		}
+		switch bitSize {
+		case 8, 16, 32:
+			return uint32(i), nil
+		case 64:
+			return i, nil
+		default:
+			return nil, fmt.Errorf("invalid bit size %d", bitSize)
+		}
+	}
+}
+
+func intHandler(bitSize int) func(interface{}) (interface{}, error) {
+	return func(s interface{}) (interface{}, error) {
+		localString := fmt.Sprintf("%v", s)
+		i, err := strconv.ParseInt(localString, 10, bitSize)
+		if err != nil {
+			return nil, err
+		}
+		switch bitSize {
+		case 8, 16, 32:
+			return int32(i), nil
+		case 64:
+			return i, nil
+		default:
+			return nil, fmt.Errorf("invalid bit size %d", bitSize)
+		}
+	}
+}
+
+func floatHandler(s interface{}) (interface{}, error) {
+	localString := fmt.Sprintf("%v", s)
+	f, err := strconv.ParseFloat(localString, 32)
+	return float32(f), err
+}
+
+func doubleHandler(s interface{}) (interface{}, error) {
+	localString := fmt.Sprintf("%v", s)
+	f, err := strconv.ParseFloat(localString, 64)
+	return f, err
+}
+
+func jsonHandler(s interface{}) (interface{}, error) {
+	localString := fmt.Sprintf("%v", s)
```

Comment:
I don't think we need to do the `unmarshal` thing here? We should just cast it to `[]byte` and return I think. Can you test this? (same as byte array)

Reply:
This still seems to be unresolved